app/javascript/react/store/movingStreamSelectors.ts

```diff
@@ -0,0 +1,138 @@
+import { createSelector } from "@reduxjs/toolkit";
+import moment, { Moment } from "moment";
+
+import { lastItemFromArray } from "../utils/lastArrayItem";
+import {
+  CalendarCellData,
+  CalendarMonthlyData,
+  StreamDailyAverage,
+} from "../types/movingStream";
+import { StreamDailyAverage as MovingStreamDailyAverage } from "../types/StreamDailyAverage";
+import { RootState } from ".";
+
+const DAYS_IN_WEEK_COUNT = 7;
+
+const getMonthWeekBoundariesForDate = (
+  date: Moment
+): { firstDayOfMonthWeek: Moment; lastDayOfMonthWeek: Moment } => {
+  const year = date.year();
+  const month = date.month();
+  const firstDayOfMonthWeek = moment([year, month])
+    .startOf("month")
+    .startOf("week");
+  const lastDayOfMonthWeek = moment([year, month]).endOf("month").endOf("week");
```

Comment:
isn't that just the last day of the month? Why the `week` at the end? Same question to `firstDayOfMonthWeek `